tests/soak-tests/src/test/java/org/apache/activemq/artemis/tests/soak/owleak/OWLeakTest.java

```diff
@@ -0,0 +1,242 @@
+      }
+
+      return parameters;
+   }
+
+   @Before
+   public void before() throws Exception {
+      Assume.assumeTrue(TEST_ENABLED);
+      cleanupData(SERVER_NAME_0);
+
+      serverProcess = startServer(SERVER_NAME_0, 0, 10_000);
+   }
+
+
+   public static void main(String[] arg) {
+      int PRODUCERS = Integer.parseInt(arg[0]);
+      int NUMBER_OF_MESSAGES = Integer.parseInt(arg[1]);
+      int MESSAGE_SIZE = Integer.parseInt(arg[2]);
+      String protocol = arg[3];
+      ExecutorService service = Executors.newFixedThreadPool(PRODUCERS + 1 + 1);
+
+      String QUEUE_NAME = "some_queue";
+
+      Semaphore semaphore = new Semaphore(PRODUCERS + 1);
+
+      CountDownLatch latch = new CountDownLatch(PRODUCERS + 1 + 1);
+
+      AtomicBoolean running = new AtomicBoolean(true);
+
+      AtomicInteger errors = new AtomicInteger(0);
+
+      try {
+
+         for (int i = 0; i < PRODUCERS; i++) {
+            final int t = i;
+            ConnectionFactory factory = CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616");
+            service.execute(() -> {
+               try {
+                  for (int msg = 0; msg < NUMBER_OF_MESSAGES; msg++) {
+                     Connection connection = factory.createConnection();
+                     Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+                     MessageProducer producer = session.createProducer(session.createQueue(QUEUE_NAME));
+                     char[] msgStr = new char[MESSAGE_SIZE];
+                     Arrays.fill(msgStr, 'a');
+                     TextMessage message = session.createTextMessage(new String(msgStr));
+                     semaphore.acquire();
+                     producer.send(message);
+                     logger.info("Thread {} Sent message with size {} with the total number of {} messages of {}", t, MESSAGE_SIZE, msg, NUMBER_OF_MESSAGES);
+                     producer.close();
+                     session.close();
+                     connection.close();
+                  }
+               } catch (Exception e) {
+                  errors.incrementAndGet();
+                  e.printStackTrace();
+                  logger.warn(e.getMessage(), e);
+               } finally {
+                  latch.countDown();
+               }
+            });
+         }
+
+
+         service.execute(() -> {
+
+            try {
+               ConnectionFactory factory = CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616");
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+               MessageConsumer consumer = session.createConsumer(session.createQueue(QUEUE_NAME));
+               connection.start();
+
+               for (int i = 0; i < NUMBER_OF_MESSAGES * PRODUCERS; i++) {
+
+                  TextMessage message = (TextMessage) consumer.receive(100_000);
```

Comment:
Seems excessive. Since so much else is configurable, perhaps make a more reasonable default and have the script/runner pass the timeout with the other settings if needing to bump it to excessive values.